tests/api/test_addon_uploads.py

```diff
@@ -1299,6 +1299,260 @@ def test_extension_invalid_addon_tags(base_url, session_auth):
+    # add a caption for all the available previews
+    for preview in previews_id:
+        edit_addon = requests.patch(
+            url=f'{base_url}{_addon_create}{addon}/previews/{preview}/',
+            headers={
+                'Authorization': f'Session {session_auth}',
+                'Content-Type': 'application/json',
+            },
+            data=json.dumps(payload),
+        )
+        response = edit_addon.json()
+        assert (
+            edit_addon.json()['caption'] == payload['caption']
+        ), f'Actual response was {response}'
+
+
+@pytest.mark.serial
+@pytest.mark.create_session('api_user')
+def test_extension_no_image_attached(base_url, session_auth):
+    """Send a screenshot upload request without adding an image"""
+    addon = payloads.edit_addon_details['slug']
+    edit_addon = requests.post(
+        url=f'{base_url}{_addon_create}{addon}/previews/',
+        headers={'Authorization': f'Session {session_auth}'},
+    )
+    assert (
+        edit_addon.status_code == 400
+    ), f'Actual status code was {edit_addon.status_code}'
+    assert (
+        'No file was submitted.' in edit_addon.text
+    ), f'Actual response message was {edit_addon.text}'
+
+
+@pytest.mark.parametrize(
+    'preview',
+    [
+        'img/bmp_icon.bmp',
+        'img/static_gif.gif',
+        'img/animated_png.png',
+        'img/invalid_image.png',
+    ],
+    ids=[
+        'BMP screenshot',
+        'GIF static screenshot',
+        'PNG animated screenshot',
+        'Non image file with a .png extension',
+    ],
+)
+@pytest.mark.serial
+@pytest.mark.create_session('api_user')
+def test_extension_add_invalid_image(base_url, session_auth, preview):
+    """Verify that requests fail if images do not meet these acceptance criteria:
+    PNG or JPG, non-animated images, valid image file"""
+    addon = payloads.edit_addon_details['slug']
+    with open(preview, 'rb') as img:
+        edit_addon = requests.post(
+            url=f'{base_url}{_addon_create}{addon}/previews/',
+            headers={'Authorization': f'Session {session_auth}'},
+            files={'image': img},
+        )
+        print(
+            f'For image "{preview}": Response status is {edit_addon.status_code}; {edit_addon.text}\n'
+        )
+        assert (
+            edit_addon.status_code == 400
+        ), f'Actual status code was {edit_addon.status_code}'
+        if preview == 'img/bmp_icon.bmp' or preview == 'img/static_gif.gif':
+            assert (
+                'Images must be either PNG or JPG.' in edit_addon.text
+            ), f'Actual response was {edit_addon.text}'
+        elif preview == 'img/animated_png.png':
+            assert (
+                'Images cannot be animated.' in edit_addon.text
+            ), f'Actual response was {edit_addon.text}'
+        else:  # for the non-image file
+            assert (
+                'Upload a valid image.' in edit_addon.text
+            ), f'Actual response was {edit_addon.text}'
```

Comment:
You could put these messages in a list and assert the ```edit_addon.text``` string matches.

Reply:
@jrbenny35 That's a good idea. I've chosen to save the messages in a list as a variable (in the `dev.json` variables file), since I'm verifying the same outcome for both addon icons and screenshots. I can now verify this variable against the response messages in both tests. I've made sure that the messages list follows the same order as the parameters (the images we loop trough in `@pytest.mark.parametrize`) which means I can make only one assertion now by using a `count` to loop through the list: `assert variables['image_validation_messages'][count] in edit_addon.text`.